modules.cmake

```diff
@@ -203,9 +203,13 @@ function(add_module_library name)
     endforeach ()
   endif ()
 
-  target_sources(${name} PRIVATE ${sources})
+  if(CMAKE_VERSION VERSION_LESS 3.28)
+    target_sources(${name} PRIVATE ${sources})
+  else()
+    target_sources(${name} PUBLIC FILE_SET fmt_module TYPE CXX_MODULES FILES ${sources})
+  endif()
```

Comment:
I think it's better to restructure the code to check CMake version in one place instead of three, something like:

```cmake
if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.28)
    target_sources(${name} PUBLIC FILE_SET fmt_module TYPE CXX_MODULES FILES ${sources})
    return()
endif ()
```
earlier in this function

Also note spaces before opening parentheses in statement-like constructs like `if`.